packages/front-end/components/Experiment/TabbedPage/StartExperimentModal.tsx

```diff
@@ -0,0 +1,116 @@
+import { ExperimentInterfaceStringDates } from "back-end/types/experiment";
+import { useState } from "react";
+import Modal from "@/components/Modal";
+import { useUser } from "@/services/UserContext";
+import UpgradeModal from "@/components/Settings/UpgradeModal";
+import PremiumCallout from "@/components/Radix/PremiumCallout";
+
+export interface Props {
+  experiment: ExperimentInterfaceStringDates;
+  close: () => void;
+  startExperiment: () => Promise<void>;
+  checklistItemsRemaining: number;
+}
+
+export default function StartExperimentModal({
+  experiment,
+  close,
+  startExperiment,
+  checklistItemsRemaining,
+}: Props) {
+  const checklistIncomplete = checklistItemsRemaining > 0;
+
+  const [upgradeModal, setUpgradeModal] = useState(false);
+
+  const { hasCommercialFeature } = useUser();
+
+  const needsVisualEditorUpgrade =
+    experiment.hasVisualChangesets && !hasCommercialFeature("visual-editor");
+
+  const needsRedirectUpgrade =
+    experiment.hasURLRedirects && !hasCommercialFeature("redirects");
+
+  const needsUpgrade = needsVisualEditorUpgrade || needsRedirectUpgrade;
+
+  if (needsUpgrade && upgradeModal) {
+    return (
+      <UpgradeModal
+        close={() => setUpgradeModal(false)}
+        commercialFeature={
+          needsVisualEditorUpgrade ? "visual-editor" : "redirects"
+        }
+        source="start-experiment-modal"
+      />
+    );
+  }
+
+  return (
+    <Modal
+      trackingEventModalType="start-experiment"
+      trackingEventModalSource={
+        checklistIncomplete ? "incomplete-checklist" : "complete-checklist"
+      }
+      open={true}
+      size="md"
+      submit={startExperiment}
+      cta="Start Now"
+      ctaEnabled={!checklistIncomplete && !needsUpgrade}
+      close={close}
+      secondaryCTA={
+        checklistIncomplete && !needsUpgrade ? (
+          <button
+            className="btn btn-link text-decoration-none"
+            onClick={async () => startExperiment()}
+          >
+            <span
+              style={{
+                color: "var(--text-color-danger)",
+              }}
+            >
+              Start Anyway
+            </span>
+          </button>
+        ) : null
+      }
+      header="Start Experiment"
+    >
+      <div className="p-2">
+        {checklistIncomplete ? (
+          <div className="alert alert-warning">
+            You have{" "}
+            <strong>
+              {checklistItemsRemaining} task
+              {checklistItemsRemaining > 1 ? "s " : " "}
+            </strong>
+            left to complete. Review the Pre-Launch Checklist before starting
+            this experiment.
+          </div>
+        ) : null}
+
+        {needsVisualEditorUpgrade ? (
+          <PremiumCallout
+            commercialFeature="visual-editor"
+            id="start-experiment-modal"
+            mb="3"
+          >
+            This experiment contains visual editor changes, which require a paid
+            plan.
```

Comment:
I'm wondering if it would add value to add a doc link at the end here that links to a doc article about how to preview visual editor / url redirect experiments without a pro/enterprise plan.

The idea being the first visual editor / url redirect experiment the org creates could be a test experiment that they launch to internal users in order to test out the functionality. When instead they should be keeping it in a draft and using the Dev Tools to preview it.

Reply:
The visual changes section has preview links, but they are easy to miss.  I think this is a general problem, not specific to this change.  It's not super clear on a draft experiment how to preview and QA the experiment before starting.

![image](https://github.com/user-attachments/assets/e0a749b7-6be6-465f-a515-b883100e945c)